napari/layers/points/_points_key_bindings.py

```diff
@@ -58,7 +60,61 @@ def paste(layer: Points):
     trans._("Select all points in the current view slice."),
 )
 def select_all(layer: Points):
-    layer.selected_data = set(layer._indices_view[: len(layer._view_data)])
+    new_selected = set(layer._indices_view[: len(layer._view_data)])
+
+    # If all visible points are already selected, deselect the visible points
+    if new_selected & layer.selected_data == new_selected:
+        layer.selected_data = layer.selected_data - new_selected
+        show_info(
+            trans._(
+                "Deselected all points in this slice,"
+                " use Shift-A to deselect all points on the layer."
+                f" ({len(layer.selected_data)} selected)"
+            )
+        )
+
+    # If not all visible points are already selected, additionally select the visible points
+    else:
+        layer.selected_data = layer.selected_data | new_selected
+        show_info(
+            trans._(
+                f"Selected {len(new_selected)} points in this slice,"
+                " use Shift-A to select all points on the layer."
+                f" ({len(layer.selected_data)} selected)"
+            )
+        )
+    layer._set_highlight()
+
+
+@register_points_action(
+    trans._("Select all points in the layer."),
+)
+def select_all_3d(layer: Points):
```

Comment:
I think 3D is incorrect here — this is nD. (One can have 3D slices in a 3D+t time series, for example.) Perhaps change the two function names to `select_all_in_slice` and `select_all_data`?

Reply:
Done